include/boost/integer/integer_log2.hpp

```diff
@@ -16,7 +16,7 @@
 #define BOOST_INTEGER_INTEGER_LOG2_HPP
 
 #include <assert.h>
-#ifdef __BORLANDC__
+#if defined(__BORLANDC__) && !defined(__clang__)
```

Comment:
Shouldn't this one become `defined(BOOST_BORLANDC)`?

Reply:
No. Boost config.hpp is not being included at that point so that I must use the long form of checking for BOOST_BORLANDC.

Reply:
`boost/config.hpp` is included just below, so this include could be moved below that one.

Reply:
Yes, but why bother. There is nothing wrong with the code as it now is.

Reply:
I thought the whole point of these macros was to remove code duplication.

Reply:
The point of the macros was to use them as shorthand when config.hpp had already been included, else use the longhand check.  The check is so that the Embarcadero C++ clang-based compilers were not included in workarounds for the old bcc32 compiler. But I have no intention of changing the order of code just so I can use the shorthand macros instead of the longhand method.

Reply:
@eldiener Now that there are two BCC compilers that ship with Embarcadero C++ Builder: BCC32X (the clang based one) and BCC32C (the one compatible with the older Borland C++ BCC32 compiler), does   `defined(__BORLANDC__) && !defined(__clang__)` apply to BCC32C too, or just ancient BCC32?  (Either way, if you're touching code in Integer which @Lastique maintains, and he would prefer consistency among all Borland compiler checks, it seems like a reasonable request).

Reply:
Embarcadero C++ has four Windows compilers. The bcc32 is the old Windows compiler and does not define __clang__, and is what all the old workaround code is all about. The bcc32c and bcc32x are 32-bit clang-based compilers, both defining __clang__ and the only difference between the two is that bcc32c takes the same command line options that the old bcc32 takes while bcc32x takes clang-based command line options. Finally their is bcc64, which is a 64-bit compiler, defining __clang__, which takes clang-based command line options. None of the clang-based compilers should need to be using the old bcc32 workarounds Boost has in its code. I have no interest in bcc32 and I really doubt that even with the workarounds in Boost code that it can currently compile more than about 5-10 percent of current Boost successfully. My only interest is to have the clang-based compilers work with current Boost on an ongoing basis and it is those compilers I test when making PRs and issuing changes. 

As far as consistency goes I have already explained that if config.hpp is included then the macros BOOST_BORLANDC, BOOST_CODEGEARC, and BOOST_EMBTC can be used, but if config.hpp has not been included than the checks for __BORLANDC__ and !__clang__, __CODEGEARC__ and !__clang__, and __CODEGEARC__ and __clang__ are just as good. I am actually trying to change code as little as possible by not moving around header files. But if he insists that I move around his code so that config.hpp is earlier in the file so that the shorthand can be used I will do it. Or of course he can do it after he merges the PR.

Reply:
But `config.hpp` is included, literally a few lines lower. Not using the macros from it, when the whole point of the PR is to port to those macros, makes no sense. Of course, I can fix it myself, and apparently I'll have to.